pkg/providers/amazon/objectstore/objectstore.go

```diff
@@ -178,7 +179,15 @@ func (s *objectStore) DeleteBucket(bucketName string) error {
 		Bucket: aws.String(bucketName),
 	}
 
-	_, err := s.client.DeleteBucket(input)
+	obj, err := s.ListObjects(bucketName)
+	if err != nil {
+		return emperror.With(emperror.Wrap(err, "could not list objects"), "bucket", bucketName)
+	}
+
+	if len(obj) > 0 {
+		return emperror.With(pkgErrors.ErrorBucketDeleteNotEmpty, "bucket", bucketName)
+	}
+	_, err = s.client.DeleteBucket(input)
 	if err != nil {
 		err = s.convertError(err)
 		return emperror.With(emperror.Wrap(err, "bucket deletion failed"), "bucket", bucketName)
```

Comment:
use emperror's WrapWith instead; the woding should be "faieled to list objects" (to be consistent across providers)